notebooks/LeNet-example.ipynb

```diff
@@ -0,0 +1,555 @@
+       "  )\n",
+       ")"
+      ]
+     },
+     "execution_count": 26,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "model = LeNet()\n",
+    "model.to(\"cuda\").eval()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Iteration 1000, ave batch time 0.91 ms\n",
+      "Iteration 2000, ave batch time 0.90 ms\n",
+      "Iteration 3000, ave batch time 0.90 ms\n",
+      "Iteration 4000, ave batch time 0.90 ms\n",
+      "Iteration 5000, ave batch time 0.90 ms\n",
+      "Iteration 6000, ave batch time 0.90 ms\n",
+      "Iteration 7000, ave batch time 0.90 ms\n",
+      "Iteration 8000, ave batch time 0.90 ms\n",
+      "Iteration 9000, ave batch time 0.90 ms\n",
+      "Iteration 10000, ave batch time 0.90 ms\n",
+      "Average batch time: 0.90 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "benchmark(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Tracing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "LeNet(\n",
+       "  original_name=LeNet\n",
+       "  (feat): LeNetFeatExtractor(\n",
+       "    original_name=LeNetFeatExtractor\n",
+       "    (conv1): Conv2d(original_name=Conv2d)\n",
+       "    (conv2): Conv2d(original_name=Conv2d)\n",
+       "  )\n",
+       "  (classifer): LeNetClassifier(\n",
+       "    original_name=LeNetClassifier\n",
+       "    (fc1): Linear(original_name=Linear)\n",
+       "    (fc2): Linear(original_name=Linear)\n",
+       "    (fc3): Linear(original_name=Linear)\n",
+       "  )\n",
+       ")"
+      ]
+     },
+     "execution_count": 29,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "traced_model = torch.jit.trace(model, torch.empty([1,1,32,32]).to(\"cuda\"))\n",
+    "traced_model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Iteration 1000, ave batch time 0.66 ms\n",
+      "Iteration 2000, ave batch time 0.66 ms\n",
+      "Iteration 3000, ave batch time 0.66 ms\n",
+      "Iteration 4000, ave batch time 0.65 ms\n",
+      "Iteration 5000, ave batch time 0.65 ms\n",
+      "Iteration 6000, ave batch time 0.65 ms\n",
+      "Iteration 7000, ave batch time 0.65 ms\n",
+      "Iteration 8000, ave batch time 0.65 ms\n",
+      "Iteration 9000, ave batch time 0.65 ms\n",
+      "Iteration 10000, ave batch time 0.65 ms\n",
+      "Average batch time: 0.65 ms\n"
+     ]
+    }
+   ],
+   "source": [
+    "benchmark(traced_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Scripting"
```

Comment:
Can we have a little section of text here explaining what is happening and the difference between script and trace paths?

Reply:
good idea. Added some explanation for script and trace now (text taken from the blog)